scanner/updater/manual/manual.go

```diff
@@ -1,39 +1,220 @@
+	FixedInVersion      string `json:"FixedInVersion"`
+	Repo                struct {
+		Name string `json:"Name"`
+		URI  string `json:"URI"`
+	} `json:"Repo"`
+}
 
-// UpdaterSet creates a new updater set with the provided vulnerability data.
-func UpdaterSet(_ context.Context) (driver.UpdaterSet, error) {
-	res := driver.NewUpdaterSet()
-	err := res.Add(&updater{})
+type Vulnerabilities struct {
+	Vulnerabilities []Vulnerability `json:"vulnerabilities"`
+}
+
+const (
+	Name = "stackrox-manual"
+	// DefaultURL Default URL to fetch the vulnerabilities JSON.
+	DefaultURL = "https://raw.githubusercontent.com/stackrox/stackrox/master/scanner/updater/manual/vulns.json"
+)
+
+var defaultClient = &http.Client{
+	Transport: proxy.RoundTripper(),
+	Timeout:   5 * time.Minute,
+}
+
+type updater struct {
+	c         *http.Client
+	updateURL *url.URL
+}
+
+// NewUpdater creates a new instance of the updater with default settings.
+func NewUpdater(c *http.Client, uri string) (*updater, error) {
+	var url *url.URL
+	var err error
+	if uri == "" {
+		uri = DefaultURL
+	}
+	url, err = url.Parse(uri)
 	if err != nil {
-		return res, err
+		return nil, err
 	}
-	return res, nil
+	return &updater{c: c, updateURL: url}, nil
+}
+
+func (u *updater) Name() string {
+	return Name
+}
+
+// Fetch fetching data from a configurable URI.
+func (u *updater) Fetch(ctx context.Context, fingerprint driver.Fingerprint) (io.ReadCloser, driver.Fingerprint, error) {
+	ctx = zlog.ContextWithValues(ctx, "component", "updater/manual/manual.Fetch")
+	resp, err := u.c.Get(u.updateURL.String())
+	if err != nil {
+		return nil, "", fmt.Errorf("failed to fetch data: %w", err)
+	}
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
+			zlog.Error(ctx).Err(err).Msg("failed to close response body")
+		}
+	}()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, "", fmt.Errorf("bad response status: %s", resp.Status)
+	}
+	out, err := os.CreateTemp("", "manual")
+	if err != nil {
+		return nil, "", errors.Wrap(err, "creating scanner defs file")
+	}
+	zlog.Debug(ctx).
+		Str("filename", out.Name()).
+		Msg("opened temporary file for output")
+
+	_ = os.RemoveAll(out.Name())
+	_, err = io.Copy(out, resp.Body)
+
+	if err != nil {
+		utils.IgnoreError(out.Close)
+		return nil, "", fmt.Errorf("failed to write to temporary file: %w", err)
+	}
+
+	if _, err = out.Seek(0, io.SeekStart); err != nil {
+		utils.IgnoreError(out.Close)
+		return nil, "", fmt.Errorf("seek failed: %w", err)
+	}
+	zlog.Info(ctx).
+		Str("dir", out.Name()).
+		Msg("fetched manual vulnerability json file")
+	return out, "", nil
+}
+
+// Parse parsing the fetched json file into vulnerabilities.
+func (u *updater) Parse(ctx context.Context, rc io.ReadCloser) ([]*claircore.Vulnerability, error) {
+	defer func() {
+		_ = rc.Close()
+	}()
+
+	decoder := json.NewDecoder(rc)
+	// Expecting the JSON object to start with an opening brace
+	if _, err := decoder.Token(); err != nil {
```

Comment:
This file size will definitely be on the smaller side, so I think we can simplify this to just read the whole thing into memory at once into a `[]Vulnerability`

Reply:
I am a bit concerned if the file may grow over time. But it may never be big enough to blow the memory?

Reply:
Put one record per line, then:

```golang
	d := json.NewDecoder(r)
	for 
		err = d.Decode(&vuln)
		if err != nil {
			break
		}
		...
	}
```

Then it's simpler code + can grow as much as we want.

(I suggested this in another comment.)

Reply:
> I am a bit concerned that the file may grow over time. But it may never be big enough to blow the memory?

I can't imagine this file will ever be massive enough to require streaming reads. Victor also offers another option that'd make this cleaner, too. Instead of making the file:

```
[
  {
    vuln1
  },
  {
    vuln2
  },
]
```

you can do:

```
{
  vuln1
},
{
  vuln2
}
```
This way you can decode it the way he suggested ^

Reply:
If we move to YAML, I suggest we load everything in memory; "this file will [never] be massive enough to require streaming reads."